src/main/java/org/mockito/internal/util/reflection/StaticFinalOverrider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.util.reflection;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+/**
+ * To set a value on a static final field, setting the field accessible
+ * by using {@link Field#setAccessible(boolean)} is not enough.
+ * Only if removing the field modifier {@link Modifier#STATIC}, a value can be set via reflection.
+ */
+public class StaticFinalOverrider {
+    private Field modifiersField = null;
+
+    public void enableWrite(Field field) {
+        if ((field.getModifiers() & (Modifier.FINAL + Modifier.STATIC)) == 0) {
+            return;
+        }
+
+        try {
+            modifiersField = Field.class.getDeclaredField("modifiers");
+            modifiersField.setAccessible(true);
+            modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+        } catch (NoSuchFieldException e) {
```

Comment:
Thinking about this, I'm not sure this always work. I know this trick, but it may not work under all circumstaces e.g. when there's a security manager.

Which means doing it in `AccessController.doPrivileged(...)`. However I suspects that a lot of mockito code base isn't complient with secuirty manager anyway.

Reply:
Hmm... setAccessible itself is guarded by the SecurityManager as per JavaDoc.
Calling setInt only throws IllegalAccessException (from Field javadoc) "if this Field object is enforcing Java language access control and the underlying field is either inaccessible or final."
Since we're setting the Field accessible, the IllegalAccessException *should* not occur.

Unfortunately I don't have any experience with SecurityManagers but do share your assumption: 
"However I suspects that a lot of mockito code base isn't complient with secuirty manager anyway." :)

Risk it to leave it as-is?